src/native.rs

```diff
@@ -190,6 +190,41 @@ impl super::Context for Context {
         }
     }
 
+    unsafe fn get_active_uniforms(&self, program: Self::Program) -> u32 {
+        let gl = &self.raw;
+        let mut count = 0;
+        gl.GetProgramiv(program, ACTIVE_UNIFORMS, &mut count);
+        count as u32
+    }
+
+    unsafe fn get_active_uniform(&self, program: Self::Program, index: u32) -> Option<ActiveUniform> {
+        let gl = &self.raw;
+        let mut uniform_max_size = 0;
+        gl.GetProgramiv(program, ACTIVE_UNIFORM_MAX_LENGTH, &mut uniform_max_size);
+
+        let mut name = String::with_capacity(uniform_max_size as usize);
```

Comment:
I know these came from gfx's implementation but I wonder if `String` is correct here (and the other couple functions that which do this), normally I'd expect `CString`/`CStr`. I can try to look into the history of this, maybe we should revisit it later

Reply:
Yeah I was wondering about that myself.  I copied this from elsewhere in glow, but I wasn't totally sure it was even sound?  I guess practically it has to be since identifiers from shaders have to be valid variable names... probably?  Even if it was though I'd also expect CString and it's not really any less useful.